### PR TITLE
this commit should show kata 6 completed

### PR DESCRIPTION
**Pull request by layla93k**

### CODE REVIEW

File: Models/models.js [hunk 1]
```diff
@@ -36,4 +36,21 @@ exports.fetchAllArticles = () => {
...
-}
\ No newline at end of file
+}
+
+exports.fetchArticleCommentsById = (article_id) => {
+    const query = `SELECT comments.comment_id, 
+    comments.votes, comments.created_at, comments.author, comments.body, comments.article_id
+    FROM comments
+    LEFT JOIN articles ON articles.article_id= comments.article_id
+    WHERE comments.article_id = $1
```
Comment (Rayhaan2k9) [inline Models/models.js:46]: There is no need for a join here as all the information you need exists on the comments table 🙂

File: __tests__/app.test.js [hunk 2]
```diff
@@ -117,55 +101,65 @@ describe('GET /api/articles', () => {
...
+                    descending: true,
+                })
+               body.comments.forEach((comment)=>{
+                expect(typeof comment.comment_id).toBe('number')
+                expect(typeof comment.votes).toBe('number')
+                expect(typeof comment.created_at).toBe('string')
+                expect(typeof comment.author).toBe('string')
+                expect(typeof comment.body).toBe('string')
+                expect(typeof comment.article_id).toBe('number')
+               })
```
Comment (Rayhaan2k9) [inline __tests__/app.test.js:143]: ```typeof``` does work here but doesn't look great! Have a look at ```toMatchObject``` instead 😄

File: __tests__/app.test.js [hunk 3]
```diff
@@ -117,55 +101,65 @@ describe('GET /api/articles', () => {
...
+            });
+    });
+    test('GET:400 sends an appropriate status and error message when given an invalid id', () => {
+        return request(app)
+            .get('/api/articles/not-an-id/comments')
+            .expect(400)
+            .then((response) => {
                 expect(response.body.msg).toBe('Bad request');
-              });
+            });
```
Comment (Rayhaan2k9) [inline __tests__/app.test.js:163]: Love these error handling tests! However, you are missing a test here. What if the article does exist, however, has no comments associated with it? For example article_id 13? This should be returning a 200 status with an empty array response

File: Models/models.js [hunk 4]
```diff
@@ -36,4 +36,21 @@ exports.fetchAllArticles = () => {
...
+    ORDER BY comments.created_at DESC;`
+    
+    return db.query(query, [article_id]).then(({rows})=>{
+        if(rows.length === 0) {
+            return Promise.reject({status: 404, msg: 'article_id does not exist'})
+          }
```
Comment (Rayhaan2k9) [inline Models/models.js:52]: You will need to add something to your model to handle valid articles that have no comments 🙂

**Review by Rayhaan2k9 — CHANGES_REQUESTED** [review]
Almost there with this, Layla! I've left some comments for you 🙂

**Review by robert112233 — APPROVED** [review]
Great work! You've included all the necessary tests and passed them appropriately! Merge 🥇

**Pull request merged**